plugin/storage/es/spanstore/reader.go

```diff
@@ -444,10 +456,17 @@ func (s *SpanReader) buildOperationNameQuery(operationName string) elastic.Query
 }
 
 func (s *SpanReader) buildTagQuery(k string, v string) elastic.Query {
-	queries := make([]elastic.Query, len(tagFieldList))
-	for i := range queries {
-		queries[i] = s.buildNestedQuery(tagFieldList[i], k, v)
+	queries := make([]elastic.Query, len(nestedTagFieldList)+len(objectTagFieldList))
+	kd := strings.Replace(k, ".", ":", -1)
+	for i := range objectTagFieldList {
+		queries[i] = s.buildObjectQuery(objectTagFieldList[i], kd, v)
+	}
+	for i := range nestedTagFieldList {
+		queries[i+len(objectTagFieldList)] = s.buildNestedQuery(nestedTagFieldList[i], k, v)
 	}
+
+	// TODO we could query tagsMap only if the tag is configured
+	// but configuration can change over time
```

Comment:
I mean what if previous deployment was storing tags as object fields. If the new deployment is not using tags as fields query would not find old tags.